addons/web_editor/static/src/js/backend/html_field.js

```diff
@@ -113,6 +115,7 @@ export class HtmlField extends Component {
                 this.currentEditingValue = undefined;
             }
             this._lastRecordInfo = newRecordInfo;
+            this.shouldReset = newProps.record._config.context.shouldReset;
```

Comment:
The use of that prop and this `_config.context` here seems hacky and overkill to me. However, I don't have a clear idea on how else it could be done. @Goaman I'm sure you can come up with a better idea! :)

Reply:
@dmo-odoo After talking with @Mtaylorr, we will explore another solution

Reply:
@dmo-odoo , @Goaman , I updated the fix and it seems to be working fine and doesn't break the behavior introduced by https://github.com/odoo/odoo/commit/0e22763a85ee2f4caf663185eb70d3d181b954b0